packages/api/src/routes/nfts-upload.js

```diff
@@ -45,19 +45,27 @@ export async function upload(event, ctx) {
     }
     nftSize = size
   } else {
-    const blob = await event.request.blob()
+    let blob = await event.request.blob()
     if (blob.size === 0) {
       throw new HTTPError('Empty payload', 400)
     }
-    const { cid, size } = await cluster.add(blob)
+    if (
+      contentType.includes('application/car') &&
+      blob.type !== 'application/car'
+    ) {
+      // force the content type on the blob without duplicating content
+      blob = blob.slice(0, blob.size, 'application/car')
+    }
+    // cluster returns `bytes` rather than `size` when upload is a CAR.
+    const { cid, size, bytes } = await cluster.add(blob)
     nft = {
       cid,
       created,
       type: blob.type,
       scope: tokenName,
       files: [],
     }
-    nftSize = size
+    nftSize = size || bytes
```

Comment:
This could lead into misleading size for the NFT, if we do not provide all the blocks, right?

Reply:
Yes, if the same code path is used for uploading partial CARs then the bytes value would likely be inaccruate